### ISSUE TITLE
[Feature] STPPaymentCardTextField zip code field launches numeric keyboard rather than numbers and punctuation

### ISSUE THREAD

**Issue by upside-sarah** [stripe/stripe-ios#2026]
## Summary
The zip code field currently defaults to the numbers and punctuation keyboard regardless of `countryCode`, although most US-based users expect to input only a 5 digit zip. This behavior is different on Android, which does launch _only_ the numeric keyboard (without punctuation) when `usZipCodeRequired = true`.

## Describe the solution you'd like
If possible, it would be great if setting the `countryCode` to `US` would default to the pure numeric keyboard (like the PAN, CVC, and expiry fields), or if a similar field (`usZipCodeRequired`) was included in the iOS SDK. Thanks!

**Comment by wooj-stripe** [comment]
Hi @upside-sarah!

Thanks for reaching out to us.  I looked into this and it seems that the keyboard type was changed to show `.numbersAndPunctuation` to allow shoppers who have billing addresses that aren't located in the US.  We found that presenting a numeric-only keyboard lead to shoppers being inadvertently unable to check out with their credit cards.

As such, updating the behavior by setting the `countryCode` to `US` or introducing a the field `usZipCodeRequired` on iOS is currently not on our roadmap, but if anything changes, we'll update this thread.
